tests/ga/test_extension.py

```diff
@@ -2322,324 +2324,320 @@ def test_it_should_parse_boolean_values_encoded_as_strings(self):
 
 
 class TestExtensionUpdateOnFailure(ExtensionTestCase):
+    
+    def setUp(self):
+        AgentTestCase.setUp(self)
+        self.mock_sleep = patch("time.sleep", lambda *_: mock_sleep(0.0001))
+        self.mock_sleep.start()
+
+    def tearDown(self):
+        self.mock_sleep.stop()
+        AgentTestCase.tearDown(self)
 
     @staticmethod
-    def _get_ext_handler_instance(name, version, handler=None, continue_on_update_failure=False):
+    def _do_upgrade_scenario(first_ext, upgraded_ext):
```

Comment:
NIT: Rename to `_do_upgrade_scenario_and_get_order` or something moer verbose for more clarity

Reply:
To clarify: the intent here is to make explicit the returned value, correct?

Reply:
Yes that is correct, the function name should be explicit enough to tell you exactly what its doing. (Feel free to choose another name if you like too, I'm not that great with names lol)